File: app/src/main/java/im/point/dotty/network/MetaPost.java
package im.point.dotty.network;

import com.google.gson.annotations.SerializedName;

public final class MetaPost {
    @SerializedName("bookmarked") boolean isBookmarked;
    long uid;
    @SerializedName("subscribed") boolean isSubscribed;
    @SerializedName("editable") boolean isEditable;
    @SerializedName("recommended") boolean isRecommended;
    Post post;

    public boolean isBookmarked() {
        return isBookmarked;
    }

    public long getUid() {
        return uid;
    }

    public boolean isSubscribed() {
        return isSubscribed;
    }

    public boolean isEditable() {
        return isEditable;
    }

    public boolean isRecommended() {
        return isRecommended;
    }

    public Post getPost() {
        return post;
    }
}

File: app/src/main/java/im/point/dotty/network/MetaPost.kt
/*
 * Copyright (c) 2019-2021 Pavel A. Sokolov
 */
package im.point.dotty.network

import com.google.gson.annotations.SerializedName

data class MetaPost(@SerializedName("bookmarked")
                    var isBookmarked: Boolean? = null,
                    var uid: Long? = null,
                    @SerializedName("subscribed")
                    var isSubscribed: Boolean? = null,
                    @SerializedName("editable")
                    var isEditable: Boolean? = null,
                    @SerializedName("recommended")
                    var isRecommended: Boolean? = null,
                    @SerializedName("comment_id")
                    var commentId: Int? = null,
                    var post: RawPost? = null) {

    val id: String
        get() = post?.id ?: throw Exception("post ID is empty")
}